### PR TITLE
Telugu courses added

### PR DESCRIPTION
**Pull request by TejaPutcha**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by TejaPutcha** [comment]
Can anyone please help me here? I do not understand what the problem is.

File: courses/free-courses-te.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+* [SQL Playlist](https://www.youtube.com/playlist?list=PLXx2-0oYp1LPUXvjjriVMaMWALucsitR1) - VLR Training
+
+
+### VueJS
+
+* [VueJS Playlist](https://www.youtube.com/playlist?list=PLO7Oa5iXf4Qjd9AjhYSkOoY4aPe1pkKzk) - TeluguTechSteps
+
+
+
+### NPTEL
```
Comment (shrn01) [inline courses/free-courses-te.md:212]: you already have a NPTEL heading, move this resource there

File: courses/free-courses-te.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+# Index
+
+## YouTube
```
Comment (shrn01) [inline courses/free-courses-te.md:3]: this heading may not be required

File: courses/free-courses-te.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+* [NodeJS Tutorial](https://www.youtube.com/watch?v=MY2Vxtfn5Tw) - Telugu Skillhub
+* [NodeJS Playlist](https://www.youtube.com/playlist?list=PLYnehuuSeAHtu27M2By66v6kJpF_oDR5I) - Know something!!!
+
+
+### PHP
+
+* [PHP Development Part-1](https://www.youtube.com/watch?v=sX6g3zyPXkA) - Telugu Computer World
+* [PHP Development Part-2](https://www.youtube.com/watch?v=zZ0QNRMxWkE) - Telugu Computer World
+* [PHP Playlist](https://www.youtube.com/playlist?list=PLC2mgeYbYNm8TRhX27z6JG0DtR8FR_WuO) - Telugu Computer World
+
```
Comment (shrn01) [inline courses/free-courses-te.md:185]: remove a newline. only 2 new lines between 2 sections are needed

File: courses/free-courses-te.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+### Ethical Hacking
+
+* [Ethical Hacking Complete Course](https://www.youtube.com/watch?v=96_znX8_4Mg) - Tech Cookie
+
+
+### ExpressJS
+
+* [ExpressJS Tutorial](https://www.youtube.com/watch?v=_jgN80P6YII) - Telugu Skillhub
+* [ExpressJS Playlist](https://www.youtube.com/playlist?list=PLxS8q3V3GDdzobKWCoXVYFsXlb5kyq4_N) - WhatsMySugesstion
+
```
Comment (shrn01) [inline courses/free-courses-te.md:131]: add a newline. 2 new lines are needed between 2 sections

File: courses/free-courses-te.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+### SQL
+
+* [SQL Course](https://www.youtube.com/watch?v=2XB5CddzEaM) - Edusoft Learning Systems - Learning Simplified
+* [SQL Playlist](https://www.youtube.com/playlist?list=PLXx2-0oYp1LPUXvjjriVMaMWALucsitR1) - VLR Training
+
+
+### VueJS
+
+* [VueJS Playlist](https://www.youtube.com/playlist?list=PLO7Oa5iXf4Qjd9AjhYSkOoY4aPe1pkKzk) - TeluguTechSteps
+
```
Comment (shrn01) [inline courses/free-courses-te.md:209]: remove a newline. only 2 new lines between 2 sections are needed

**Comment by TejaPutcha** [comment]
@shrn01 I made the changes.

**Comment by TejaPutcha** [comment]
Out of the 2 checks, one is failing and I don't know why. If anyone can help me here, it would be much appreciated.

File: courses/free-courses-te.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+* [Machine Learning](#machine-learning)
+* [MongoDB](#mongodb)
+* [NextJS](#nextjs)
+* [NodeJS](#nodejs)
+* [PHP](#php)
+* [Python](#python)
+* [ReactJS](#reactjs)
+* [SQL](#sql)
+* [VueJs](#vuejs)
+
```
Comment (shrn01) [inline courses/free-courses-te.md:32]: Add another new line here and you are good to go

Comment (TejaPutcha) [reply courses/free-courses-te.md:32]: Yeah, I just did that. Thank you so much! I did not know that there has to be uniformity in the blank lines that I had left. Actually, it kinda feels demotivating that the number of blank lines matters.

**Comment by TejaPutcha** [comment]
Thanks a lot @shrn01 for helping me out!

File: courses/free-courses-te.md (hunk 6 above)
Comment (shrn01) [reply courses/free-courses-te.md:32]: Hmm, actually it's in the contributing.md file. Always look into the contributing and code of conduct before contributing to open source.

Comment (TejaPutcha) [reply courses/free-courses-te.md:32]: Oh, I did not know that. Thanks for the tip will definitely follow it 👍

File: courses/free-courses-te.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+### Android
+
+* [Android App Dev](https://www.youtube.com/watch?v=7ZLKpN8vXLo) - Python Life
+* [Android Playlist](https://www.youtube.com/playlist?list=PLeCpoxUq3EhmgHp7KI-Ih3X2PcNYRtC5y) - Sai Gopi
+
+
+### Angular
+
+* [Angular For Beginners](https://www.youtube.com/watch?v=9MxS8oNlnMM) - TeluguTechSteps
```
Comment (eshellman) [inline courses/free-courses-te.md:47]: Don't invent titles! Use the title given by the resource. You can edit, but leave at least the first part of the title.

File: courses/free-courses-te.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+* [Angular Playlist](https://www.youtube.com/playlist?list=PLO7Oa5iXf4QhtPXkaNX05qhGQSKFsvAF7) - TeluguTechSteps
```
Comment (eshellman) [inline courses/free-courses-te.md:48]: Don't invent titles! Use the title given by the resource.

**Review by eshellman — CHANGES_REQUESTED** [review]
Great effort!
Please restore the books list.
you'll need to go back and get the titles from the resources, `Whatever Playlist` in not the title.

File: courses/free-courses-te.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+* [NextJS](#nextjs)
+* [NodeJS](#nodejs)
+* [PHP](#php)
+* [Python](#python)
+* [ReactJS](#reactjs)
+* [SQL](#sql)
+* [VueJs](#vuejs)
+
+
+## NPTEL
```
[the remark this answers is not in the document]
Comment (TejaPutcha) [reply courses/free-courses-te.md:34]: It is a site where courses can be done. Apart from that, reading material is provided. Yeah, I should have added it to the index. I guess it can be used both in the books file and here.

File: courses/free-courses-te.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+
+* [Various Courses in Telugu](https://sites.google.com/nptel.iitm.ac.in/translated-ebook/telugu)
```
[the remark this answers is not in the document]
Comment (TejaPutcha) [reply courses/free-courses-te.md:36]: I deleted it because I thought the file was the reason I had errors in Actions. I'll create the file and add NPTEL to it 👍

File: courses/free-courses-te.md (hunk 7 above)
Comment (TejaPutcha) [reply courses/free-courses-te.md:47]: Ok got it and on it 💯

**Comment by TejaPutcha** [comment]
I made the changes @eshellman

**Comment by TejaPutcha** [comment]
@eshellman can you please let me know what I have to do?

**Comment by TejaPutcha** [comment]
Hey, my branch has a merge conflict and I do not know how to resolve it. Can you help @shrn01 @eshellman ?

**Comment by TejaPutcha** [comment]
I resolved the merge conflict 👍

File: courses/free-courses-te.md [hunk 11]
```diff
@@ -1,39 +1,222 @@
...
+### SQL
+
+* [SQL commands and PL/SQL programs complete in telugu - Oracle SQL-PL/SQL in telugu](https://www.youtube.com/watch?v=2XB5CddzEaM) - Edusoft Learning Systems - Learning Simplified
+* [Sql tutorials in telugu - sql video tutorials - sql tutorials for beginners telugu](https://www.youtube.com/playlist?list=PLXx2-0oYp1LPUXvjjriVMaMWALucsitR1) - VLR Training
+
+
+### Tailwind CSS
+
+* [How to make Responsive Website in 5 minutes using Tailwind CSS | TELUGU | UNQCODER](https://www.youtube.com/watch?v=YVaqIIlVr7s) - UNQ Coder
+* [tailwindcss telugu | tailwind css telugu | tailwind css tutorial telugu | tailwind tutorial telugu](https://www.youtube.com/watch?v=XSI6XCgICCs) - Learn With Loks
```
Comment (shrn01) [inline courses/free-courses-te.md:217]: This is only for 19 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 12]
```diff
@@ -1,39 +1,222 @@
...
+
+### SQL
+
+* [SQL commands and PL/SQL programs complete in telugu - Oracle SQL-PL/SQL in telugu](https://www.youtube.com/watch?v=2XB5CddzEaM) - Edusoft Learning Systems - Learning Simplified
+* [Sql tutorials in telugu - sql video tutorials - sql tutorials for beginners telugu](https://www.youtube.com/playlist?list=PLXx2-0oYp1LPUXvjjriVMaMWALucsitR1) - VLR Training
+
+
+### Tailwind CSS
+
+* [How to make Responsive Website in 5 minutes using Tailwind CSS | TELUGU | UNQCODER](https://www.youtube.com/watch?v=YVaqIIlVr7s) - UNQ Coder
```
Comment (shrn01) [inline courses/free-courses-te.md:216]: This is only for 12 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 13]
```diff
@@ -1,39 +1,222 @@
...
-* [Python in Telugu - Step by Step Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-3aTUq98pbmrA3P1_m-aJR) - Telugu Computer World (YouTube)
+* [Python in Telugu For Beginners - Complete Tutorial in 10 Hours](https://www.youtube.com/watch?v=fP9IvI4qu80) - Telugu Computer World
+* [Full Python Tutorial in Telugu | Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2JgeekeyLRQcwZsfLNbW8zQ) - Telugu Web Guru
 
 
 ### R
 
-* [R Programming Language Training Videos In telugu - R ట్రైనింగ్ వీడియోస్ ఇన్ తెలుగు R programming ట్రైనింగ్ వీడియోస్ ఇన్ తెలుగు](https://www.youtube.com/playlist?list=PLXx2-0oYp1LO9H8ciGQaTr6SN80dteTlc) - VLR Training (YouTube)
+* [R Programming Language Training Videos In telugu - R ట్రైనింగ్ వీడియోస్ ఇన్ తెలుగు R programming ట్రైనింగ్ వీడియోస్ ఇన్ తెలుగు](https://www.youtube.com/playlist?list=PLXx2-0oYp1LO9H8ciGQaTr6SN80dteTlc) - VLR Training
+* [What is R programming in telugu](https://www.youtube.com/watch?v=02CTDv02awE) - Python Life
```
Comment (shrn01) [inline courses/free-courses-te.md:199]: This is only for 10 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 14]
```diff
@@ -1,39 +1,222 @@
...
+
+
+### Ethical Hacking
+
+* [Complete Ethical Hacking Course in Telugu || Tech Cookie](https://www.youtube.com/watch?v=96_znX8_4Mg) - Tech Cookie
+
+
+### ExpressJS
+
+* [Express JS in Telugu | express.js Tutorial for Beginners in Telugu](https://www.youtube.com/watch?v=_jgN80P6YII) - Telugu Skillhub
```
Comment (shrn01) [inline courses/free-courses-te.md:131]: This is only for 28 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 15]
```diff
@@ -1,39 +1,222 @@
...
+* [Data Structures in Telugu in 7hrs | Full Course | Learn Data Structures](https://www.youtube.com/watch?v=pm_ugbO2FlY) - Believer 01
+* [Data Structures](https://www.youtube.com/playlist?list=PLJSrGkRNEDAgmq4kKkPuh8aFJs-zxVbWK) - Lab Mug
 
-### <a id="cpp"></a>C++
 
-* [C++ Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm9keJjsA95jKa4EUVLd7mQP) - Telugu Computer World (YouTube)
+### Deep Learning
+
+* [Deep learning in Telugu - 1 | Deep learning tutorials in Telugu | Jp NaN Deep learning](https://www.youtube.com/watch?v=nya9vUaXAGY) - Jp NaN
+* [Neural Network in Deep Learning](https://www.youtube.com/watch?v=hfclXCz7l6U) - Python Life
```
Comment (shrn01) [inline courses/free-courses-te.md:121]: This is only for 24 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 16]
```diff
@@ -1,39 +1,222 @@
...
 
+* [Data Structures in Telugu in 7hrs | Full Course | Learn Data Structures](https://www.youtube.com/watch?v=pm_ugbO2FlY) - Believer 01
+* [Data Structures](https://www.youtube.com/playlist?list=PLJSrGkRNEDAgmq4kKkPuh8aFJs-zxVbWK) - Lab Mug
 
-### <a id="cpp"></a>C++
 
-* [C++ Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm9keJjsA95jKa4EUVLd7mQP) - Telugu Computer World (YouTube)
+### Deep Learning
+
+* [Deep learning in Telugu - 1 | Deep learning tutorials in Telugu | Jp NaN Deep learning](https://www.youtube.com/watch?v=nya9vUaXAGY) - Jp NaN
```
Comment (shrn01) [inline courses/free-courses-te.md:120]: This is only for 19 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 17]
```diff
@@ -1,39 +1,222 @@
...
+### Database Management Systems
+
+* [Data Base Management System In Telugu in 7hrs | DBMS in telugu | MySql Full Course](https://www.youtube.com/watch?v=nVgLiJOI2U8) - Believer 01
+* [DBMS Tutorial](https://www.youtube.com/playlist?list=PL3KKfF5A0sSLnIMTfr7bBw_wRW2vCm3T6) - Telugu Scit Tutorials
+
+
+### Data Science
+
+* [Data Science Course For Beginners in Telugu](https://www.youtube.com/watch?v=WKHlx--15_I) - Python Life
+* [Maths for DS](https://www.youtube.com/watch?v=8Y982AW4KuY) - Vamsi Bhavani
```
Comment (shrn01) [inline courses/free-courses-te.md:109]: This is only for 13 mins. We dont allow tutorials < 1hr in length. Please remove

File: courses/free-courses-te.md [hunk 18]
```diff
@@ -1,39 +1,222 @@
...
+
+### C++
+
+* [C++ Complete Tutorial](https://www.youtube.com/watch?v=uZBXKmQH5u8) - Telugu Computer World
+* [C++ Programming in Telugu - Complete Tutorial in 14 Hours](https://www.youtube.com/playlist?list=PLC2mgeYbYNm9keJjsA95jKa4EUVLd7mQP) - Telugu Computer World
+
+
+### Cloud Computing
+
+* [Introduction to Cloud Computing in Telugu | Cloud Computing Tutorial in Telugu | Edureka Telugu](https://www.youtube.com/watch?v=hcaROQRJM6k) - edureka! Telugu
```
Comment (shrn01) [inline courses/free-courses-te.md:85]: This is only for 19 mins. We dont allow tutorials < 1hr in length. Please remove

**Review by shrn01 — CHANGES_REQUESTED** [review]

**Comment by TejaPutcha** [comment]
I removed the courses that are less than an hour @shrn01

**Comment by TejaPutcha** [comment]
Is the PR ready to get merged @shrn01 ?

File: README.md [hunk 19]
```diff
@@ -157,6 +157,7 @@ This project lists books and other resources grouped by genres:
 + [Swedish / svenska](courses/free-courses-sv.md)
 + [Telugu / తెలుగు](courses/free-courses-te.md)
```
Comment (eshellman) [inline README.md:158]: looks like we need to fix the duplicate link

File: books/free-programming-books-te.md [hunk 20]
```diff
@@ -1,8 +1,14 @@
...
 * [C](#c)
+* [NPTEL](#nptel)
 
 
 ### <a id="c"></a>C
 
 * [Introduction to C \| Telugu](https://www.computerintelugu.com/2012/11/cmenu.html) - Sivanaadh Baazi Karampudi
+
+
+### NPTEL
```
Comment (eshellman) [inline books/free-programming-books-te.md:12]: NPTEL is not an appropriate subject heading - it's not a subject AFAICT

File: courses/free-courses-te.md [hunk 21]
```diff
@@ -1,39 +1,204 @@
...
+
+* [Android](#android)
+* [Angular](#angular)
+* [Artificial Intelligence](#artificial-intelligence)
+* [Automata Theory](#automata-theory)
+* [Bash and Shell](#bash-and-shell)
+* [Bootstrap](#bootstrap)
 * [C](#c)
-* [C++](#cpp)
+* [C++](#c-1)
```
Comment (eshellman) [inline courses/free-courses-te.md:10]: wrong. look at the anchor

File: courses/free-courses-te.md [hunk 22]
```diff
@@ -1,39 +1,204 @@
...
+* [Cloud Computing](#cloud-computing)
+* [Compiler Design](#compiler-design)
+* [CSS](#css)
```
Comment (eshellman) [inline courses/free-courses-te.md:13]: we usuallu group HTML and CSS together

File: courses/free-courses-te.md [hunk 23]
```diff
@@ -1,39 +1,204 @@
...
+* [Data Structures](#data-structures)
+* [Ethical Hacking](#ethical-hacking)
+* [ExpressJS](#expressjs)
+* [HTML](#html)
 * [Java](#java)
 * [Javascript](#javascript)
+* [Laravel](#laravel)
+* [Machine Learning](#machine-learning)
+* [MongoDB](#mongodb)
+* [NextJS](#nextjs)
```
Comment (eshellman) [inline courses/free-courses-te.md:25]: Put Javascript frameworks under javascript

File: courses/free-courses-te.md [hunk 24]
```diff
@@ -1,39 +1,204 @@
...
-* [C++](#cpp)
+* [C++](#c-1)
+* [Cloud Computing](#cloud-computing)
+* [Compiler Design](#compiler-design)
+* [CSS](#css) 
+* [Database Management Systems](#database-management-systems)
+* [Data Science](#data-science)
+* [Data Structures](#data-structures)
+* [Ethical Hacking](#ethical-hacking)
+* [ExpressJS](#expressjs)
```
Comment (eshellman) [inline courses/free-courses-te.md:18]: under javascript

File: courses/free-courses-te.md [hunk 25]
```diff
@@ -1,39 +1,204 @@
...
+* [HTML](#html)
 * [Java](#java)
 * [Javascript](#javascript)
+* [Laravel](#laravel)
+* [Machine Learning](#machine-learning)
+* [MongoDB](#mongodb)
+* [NextJS](#nextjs)
+* [NodeJS](#nodejs)
```
Comment (eshellman) [inline courses/free-courses-te.md:26]: under javascript

File: courses/free-courses-te.md [hunk 26]
```diff
@@ -1,39 +1,204 @@
...
+* [PHP](#php)
 * [Python](#python)
 * [R](#r)
+* [ReactJS](#reactjs)
```
Comment (eshellman) [inline courses/free-courses-te.md:30]: under javascript

File: courses/free-courses-te.md [hunk 27]
```diff
@@ -1,39 +1,204 @@
...
+* [Angular In Telugu | ANGULAR 10 IN TELUGU |angular in telugu|ANGULAR INTRODUCTION FOR BEGINNERS(2021)](https://www.youtube.com/watch?v=9MxS8oNlnMM) - TeluguTechSteps
+* [Angular In Telugu](https://www.youtube.com/playlist?list=PLO7Oa5iXf4QhtPXkaNX05qhGQSKFsvAF7) - TeluguTechSteps
+
+
+### Artificial Intelligence
+
+* [Artificial Intelligence 12 Hours Course In Telugu](https://www.youtube.com/watch?v=fwMMBaXIpqQ) - Python Life
+
+
+### Automata Theory
```
Comment (eshellman) [inline courses/free-courses-te.md:52]: It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category.

File: courses/free-courses-te.md [hunk 28]
```diff
@@ -1,39 +1,204 @@
...
+* [DBMS Tutorial](https://www.youtube.com/playlist?list=PL3KKfF5A0sSLnIMTfr7bBw_wRW2vCm3T6) - Telugu Scit Tutorials
+
+
+### Data Science
 
+* [Data Science Course For Beginners in Telugu](https://www.youtube.com/watch?v=WKHlx--15_I) - Python Life
 
-### <a id="cpp"></a>C++
 
-* [C++ Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm9keJjsA95jKa4EUVLd7mQP) - Telugu Computer World (YouTube)
```
Comment (eshellman) [inline courses/free-courses-te.md:18]: why is this deleted

File: courses/free-courses-te.md [hunk 29]
```diff
@@ -1,39 +1,204 @@
...
+### Data Structures
```
Comment (eshellman) [inline courses/free-courses-te.md:108]: a more common heading is data Structures and Algorithms

File: courses/free-courses-te.md [hunk 30]
```diff
@@ -1,39 +1,204 @@
...
+
+* [Data Structures in Telugu in 7hrs | Full Course | Learn Data Structures](https://www.youtube.com/watch?v=pm_ugbO2FlY) - Believer 01
+* [Data Structures](https://www.youtube.com/playlist?list=PLJSrGkRNEDAgmq4kKkPuh8aFJs-zxVbWK) - Lab Mug
+
+
+### Ethical Hacking
```
Comment (eshellman) [inline courses/free-courses-te.md:114]: It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category.

File: courses/free-courses-te.md [hunk 31]
```diff
@@ -1,39 +1,204 @@
...
+
+### HTML
+
+* [HTML Tutorials in Telugu || with in "3:30 Hours" || Computersadda.com](https://www.youtube.com/watch?v=cS0TG1iksLM) - Computers adda
+* [HTML in Telugu || HTML in Telugu by Kotha Abhishek](https://www.youtube.com/playlist?list=PLv_sM9ZH4RUWkdiiILVHnNZUsOr2DBS7S) - Chintu Tutorials
 
 
 ### Java
 
-* [Full Java Tutorials in Telugu - Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2Lyt9-2hhRj37otchec1OJL) - (YouTube)
```
Comment (eshellman) [inline courses/free-courses-te.md:23]: why did you delete this?

File: courses/free-courses-te.md [hunk 32]
```diff
@@ -1,39 +1,204 @@
...
+* [Database Management Systems](#database-management-systems)
+* [Data Science](#data-science)
+* [Data Structures](#data-structures)
+* [Ethical Hacking](#ethical-hacking)
+* [ExpressJS](#expressjs)
+* [HTML](#html)
 * [Java](#java)
 * [Javascript](#javascript)
+* [Laravel](#laravel)
+* [Machine Learning](#machine-learning)
```
Comment (eshellman) [inline courses/free-courses-te.md:23]: isn't this together with Artificial Intelligence?

File: courses/free-courses-te.md [hunk 33]
```diff
@@ -1,39 +1,204 @@
...
+
+### Bootstrap
+
+* [BOOTSTRAP INTRODUCTION IN TELUGU](https://www.youtube.com/watch?v=1MrR6eTQYgI) - TeluguTechSteps
+* [Bootstrap Tutorials in Telugu by Kotha Abhishek](https://www.youtube.com/playlist?list=PLv_sM9ZH4RUU9Yopth1p1DbEu1ebr1vTb) - Chintu Tutorials
 
 
 ### C
 
-* [C Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-n8Iz-_3MuNsJFzr0UlGUu) - Telugu Computer World (YouTube)
```
Comment (eshellman) [inline courses/free-courses-te.md:13]: Why did you delete this

File: courses/free-courses-te.md [hunk 34]
```diff
@@ -1,39 +1,204 @@
...
+### PHP
+
+* [PHP Development Tutorial in Telugu - Learn in 24 Hours Part 1](https://www.youtube.com/watch?v=sX6g3zyPXkA) - Telugu Computer World
+* [PHP Development Tutorial in Telugu - Learn in 24 Hours Part 2](https://www.youtube.com/watch?v=zZ0QNRMxWkE) - Telugu Computer World
+* [PHP Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm8TRhX27z6JG0DtR8FR_WuO) - Telugu Computer World
 
 
 ### Python
 
-* [Python in Telugu - Step by Step Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-3aTUq98pbmrA3P1_m-aJR) - Telugu Computer World (YouTube)
```
Comment (eshellman) [inline courses/free-courses-te.md:33]: Have some respect for other contributors!!!!

**Review by eshellman — CHANGES_REQUESTED** [review]

File: books/free-programming-books-te.md [hunk 35]
```diff
@@ -1,8 +1,14 @@
...
+
+* [Books - Telugu](https://sites.google.com/nptel.iitm.ac.in/translated-ebook/telugu)
```
Comment (shrn01) [inline books/free-programming-books-te.md:14]: Please add a book link directly instead of adding a repository. Remove this and add any of the books that are free and good

File: courses/free-courses-te.md [hunk 36]
```diff
@@ -1,40 +1,210 @@
...
+* [PHP Development Tutorial in Telugu - Learn in 24 Hours Part 2](https://www.youtube.com/watch?v=zZ0QNRMxWkE) - Telugu Computer World
+* [PHP Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm8TRhX27z6JG0DtR8FR_WuO) - Telugu Computer World
 
 
 ### Python
 
+* [Full Python Tutorial in Telugu | Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2JgeekeyLRQcwZsfLNbW8zQ) - Telugu Web Guru
+* [Python in Telugu - Step by Step Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-3aTUq98pbmrA3P1_m-aJR) - Telugu Computer World
+* [Python in Telugu For Beginners - Complete Tutorial in 10 Hours](https://www.youtube.com/watch?v=fP9IvI4qu80) - Telugu Computer World
 * [Free Programming Fundamentals Tutorial - programming బిగినర్స్ ప్రోగ్రామింగ్ in telugu తెలుగు python - Udemy](https://www.udemy.com/course/programming-for-kids-in-telugu) - Saarvani R (Udemy)
```
Comment (shrn01) [inline courses/free-courses-te.md:198]: Order the resources in alphabetical order

File: courses/free-courses-te.md [hunk 37]
```diff
@@ -1,40 +1,210 @@
...
 
 ### C
 
-* [C Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-n8Iz-_3MuNsJFzr0UlGUu) - Telugu Computer World (YouTube)
+* [C Language in Telugu - Complete Tutorial in 12 Hours](https://www.youtube.com/watch?v=HdvRHC5TiwE) - Telugu Computer World
+* [C Language Tutorial](https://www.youtube.com/playlist?list=PL3KKfF5A0sSKZutcrUiTCQDX0hn7Tw61E) - Telugu Scit Tutorials
+* [C Language Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-n8Iz-_3MuNsJFzr0UlGUu) - Telugu Computer World
+
+
+### <a id = "cpp">C++</a>
```
Comment (shrn01) [inline courses/free-courses-te.md:77]:
```suggestion
### <a id="cpp"></a>C++
```

File: courses/free-courses-te.md [hunk 38]
```diff
@@ -1,40 +1,210 @@
...
+
+### JavaScript
+
+* [Full Java Script Tutorials in Telugu - Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2KqDjTuU_qHr-tI_CHOkIsN) - telugu web guru
+* [JavaScript Complete Tutorials In Telugu by Kotha Abhishek](https://www.youtube.com/watch?v=GuahuUTSUKI) - Chintu Tutorials
+* [JAVA SCRIPT FOR BEGINNERS IN 7 HOURS || LEARN JAVA SCRIPT IN 7 HOURS || JAVA SCRIPT](https://www.youtube.com/watch?v=BTuCzffKh8E) - Sundeep Saradhi Kanthety
+* [JavaScript in Telugu || JavaScript in Telugu by Kotha Abhishek](https://www.youtube.com/playlist?list=PLv_sM9ZH4RUW_Pgz-6B0Q-YTfWvC7RVFN) - Chintu Tutorials
+
+
+    #### ExpressJS
```
Comment (shrn01) [inline courses/free-courses-te.md:144]:
```suggestion
#### ExpressJS
```

File: courses/free-courses-te.md [hunk 39]
```diff
@@ -1,40 +1,210 @@
...
+
+    * [Express JS In Telugu](https://www.youtube.com/playlist?list=PLxS8q3V3GDdzobKWCoXVYFsXlb5kyq4_N) - WhatsMySugesstion
+    * [Express JS in Telugu | express.js Tutorial for Beginners in Telugu](https://www.youtube.com/watch?v=_jgN80P6YII) - Telugu Skillhub
+
+
+    #### NextJS
```
Comment (shrn01) [inline courses/free-courses-te.md:150]:
```suggestion
#### NextJS
```

File: courses/free-courses-te.md [hunk 40]
```diff
@@ -1,40 +1,210 @@
...
+    
+    * [1 What is Next js In Telugu | next js | btech in telugu](https://www.youtube.com/watch?v=9jcX6w1xHJY) - B TECH IN TELUGU
+    * [Next.js Crash Course in Telugu | Next.js in Telugu](https://www.youtube.com/watch?v=yqJlmkgroik) - Telugu Skillhub
+
 
+    #### NodeJS
```
Comment (shrn01) [inline courses/free-courses-te.md:156]:
```suggestion
#### NodeJS
```

File: courses/free-courses-te.md [hunk 41]
```diff
@@ -1,40 +1,210 @@
...
 
-### Javascript
+    * [NodeJS Tutorial](https://www.youtube.com/watch?v=MY2Vxtfn5Tw) - Telugu Skillhub
+    * [Node JS in Telugu | Node.js Tutorial for Beginners in Telugu](https://www.youtube.com/playlist?list=PLYnehuuSeAHtu27M2By66v6kJpF_oDR5I) - Know something!!!
```
Comment (shrn01) [inline courses/free-courses-te.md:159]: Please remove this empty space before the * for each resource

**Review by shrn01 — CHANGES_REQUESTED** [review]

**Review by eshellman — CHANGES_REQUESTED** [review]
Please address remaining issues

File: courses/free-courses-te.md [hunk 42]
```diff
@@ -34,22 +139,86 @@
...
+* [PHP Development Tutorial in Telugu - Learn in 24 Hours Part 1](https://www.youtube.com/watch?v=sX6g3zyPXkA) - Telugu Computer World
+* [PHP Development Tutorial in Telugu - Learn in 24 Hours Part 2](https://www.youtube.com/watch?v=zZ0QNRMxWkE) - Telugu Computer World
+* [PHP Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm8TRhX27z6JG0DtR8FR_WuO) - Telugu Computer World
 
 
 ### Python
 
 * [Free Programming Fundamentals Tutorial - programming బిగినర్స్ ప్రోగ్రామింగ్ in telugu తెలుగు python - Udemy](https://www.udemy.com/course/programming-for-kids-in-telugu) - Saarvani R (Udemy)
-* [Python in Telugu - Step by Step Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-3aTUq98pbmrA3P1_m-aJR) - Telugu Computer World (YouTube)
+* [Full Python Tutorial in Telugu | Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2JgeekeyLRQcwZsfLNbW8zQ) - Telugu Web Guru
```
Comment (eshellman) [inline courses/free-courses-te.md:208]:
```suggestion
* [Python in Telugu - Step by Step Tutorials](https://www.youtube.com/playlist?list=PLC2mgeYbYNm-3aTUq98pbmrA3P1_m-aJR) - Telugu Computer World
* [Full Python Tutorial in Telugu | Telugu Web Guru](https://www.youtube.com/playlist?list=PLh6Yk2rpZu2JgeekeyLRQcwZsfLNbW8zQ) - Telugu Web Guru
```

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books! Such an extensive PR needs a lot of review, but worth it.
